packages/dds/tree/src/feature-libraries/object-forest/objectForest.ts

```diff
@@ -70,6 +70,20 @@ export class ObjectForest extends SimpleDependee implements IEditableForest {
         recordDependency(this.dependent, this.schema);
     }
 
+    clone(schema: StoredSchemaRepository, anchors: AnchorSet): ObjectForest {
+        const forest = new ObjectForest(schema, anchors);
+        // Deep copy the trees.
+        for (const [key, value] of this.roots.fields) {
+            // TODO: this references the existing TreeValues instead of copying them:
+            // they are assumed to be copy on write. See TODO on NodeData.
+            forest.roots.fields.set(
+                key,
+                value.map((v) => mapTreeFromCursor(singleMapTreeCursor(v))),
```

Comment:
I don't quite understand what's happening here. Why is this round trip necessary? It doesn't seem very good for perf.

Reply:
When copying a forest, we have to copy the nodes so modifying one does not modify the other. Using the cursor API like this is currently the simplest way to copy MapTrees.

This forest implementation is intended to be simple, not optimized. A more optimized forest implementation might do ref counting and lazy copy on write, or have dedicated code paths for cloning its internal data structures.